src/extension/commandPaletteHandler.ts

```diff
@@ -115,4 +155,31 @@ export class CommandPaletteHandler {
             });
         });
     }
+
+    /**
+     * Publish project to exponent server. In order to do this we need to make sure the user is logged in exponent and the packager is running.
+     */
+    private executePublishToExpHost(): Q.Promise<boolean> {
+        Log.logMessage("Publishing app to Exponent server. This might take a moment.");
+        return Q(XDL.User.getCurrentUserAsync()).then(user => {
+            if (!user) {
+                Log.logWarning("You're not logged in to exponent. Please login before trying to publish.");
+                return false;
+            }
+            Log.logMessage(`Publishing as ${user.username}...`);
+            return this.startExponentPackager()
+                .then(() =>
+                    XDL.Project.publishAsync(this.workspaceRoot))
+                .then(response => {
+                    if (response.err || !response.url) {
+                        return false;
+                    }
+                    Log.logMessage(`App successfully published to ${response.url}`);
```

Comment:
We might want to do something to make this pop out and be more visible, since I think right now it's potentially in a big stream of packager output and other messages.